vega_sim/scenario/common/agents.py

```diff
@@ -2756,11 +2774,66 @@ def step(self, vega_state: VegaState):
                     trades=market_trades,
                 )
             )
+
+            if self.platform == "Linux":
+                mem_vega = (
+                    self.process_map["vega"].memory_full_info()
+                    if self.process_map["vega"] is not None
+                    else 0
+                )
+                mem_datanode = (
+                    self.process_map["data-node"].memory_full_info()
+                    if self.process_map["data-node"] is not None
+                    else 0
+                )
+            elif self.platform == "Darwin":
+                mem_vega = (
+                    self.process_map["vega"].memory_info()
+                    if self.process_map["vega"] is not None
+                    else 0
+                )
+                mem_datanode = (
+                    self.process_map["data-node"].memory_info()
+                    if self.process_map["data-node"] is not None
+                    else 0
+                )
+            else:
+                mem_vega = None
+                mem_datanode = None
+                logging.warning(
+                    "Unable to record memory usage, unsupported operating system"
+                )
+
+            self.resources.append(
+                ResourceData(
+                    at_time=start_time,
+                    vega_cpu_per=self.process_map["vega"].cpu_percent(),
+                    vega_mem_rss=mem_vega.rss if mem_vega is not None else 0,
+                    vega_mem_vms=mem_vega.vms if mem_vega is not None else 0,
+                    datanode_cpu_per=self.process_map["data-node"].cpu_percent(),
+                    datanode_mem_rss=mem_datanode.rss
+                    if mem_datanode is not None
+                    else 0,
+                    datanode_mem_vms=mem_datanode.vms
+                    if mem_datanode is not None
+                    else 0,
+                )
+            )
         if self.additional_state_fn is not None:
             self.additional_states.append(
                 self.additional_state_fn(self.vega, self.agents)
             )
 
+    def _find_processes(self):
+        self.process_map.keys()
+        for p in psutil.process_iter():
+            if (p.name() in self.process_map.keys()) and (
+                p.status() in ["running", "sleeping"]
+            ):
```

Comment:
I wonder whether there's something more robust we can do for this. This is currently going to break if someone ever runs two processes/there are leftover processes. But maybe that just has to be a caveat emptor for now.

Reply:
Also I think we'd be better having the process map start empty then have the keys specified here directly, slightly clearer what it's for and avoids a case of `None`s in the dictionary being left over

Reply:
We could use the `cmdline` property to filter out any processes not started by a `VegaServiceNull` object?

Any other left-over "vega" and "data-node" processes started by other `VegaServiceNull` objects **_should_** be either `stopped` / `zombie` / `killed` by this point.

But yeah if we ever go for a multi-node setup in market-sim we would need to expand this to monitor all nodes.